### PR TITLE
fix a11y issues in data connections modal and deploy models modal

### PR DESCRIPTION
**Pull request by jenny-s51**
**Do not merge/review, needs rebase**
Towards #965 

## Description

Delete data connections modal - before:
Screen Shot 2023-03-06 at 4 57 20 PM

Delete data connections modal - after:
Screen Shot 2023-03-06 at 5 12 56 PM

Configure model server modal - before:
Screen Shot 2023-03-06 at 5 04 26 PM

Configure model server modal - after:
Screen Shot 2023-03-06 at 5 31 49 PM

Deploy model modal - before:
Screen Shot 2023-03-09 at 10 55 34 AM
Screen Shot 2023-03-09 at 10 54 06 AM

Deploy model modal - after:
Screen Shot 2023-03-09 at 10 57 21 AM
Screen Shot 2023-03-09 at 10 54 23 AM

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

File: frontend/src/pages/projects/screens/projects/EmptyProjects.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ const EmptyProjects: React.FC = () => {
       <EmptyStateSecondaryActions>
         <Button
           href="/notebookController"
+          component="a"
```
Comment (lucferbux) [inline frontend/src/pages/projects/screens/projects/EmptyProjects.tsx:30]: So every button dedicated to navigation should have the component prop to a hyperlink right?

File: frontend/src/pages/projects/screens/detail/DetailsSection.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ const DetailsSection: React.FC<DetailsSectionProps> = ({
       <StackItem>
         <Flex>
           <FlexItem>
-            <Title id={id} headingLevel="h4" size="xl">
+            <Title id={`${id}-title`} headingLevel="h4" size="xl">
```
Comment (lucferbux) [inline frontend/src/pages/projects/screens/detail/DetailsSection.tsx:64]: Why didnt't the id worked here? Is it cause it's an enum literal?

File: frontend/src/pages/projects/screens/detail/DetailsSection.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 import { ProjectSectionID } from './types';
 
 type DetailsSectionProps = {
-  id: ProjectSectionID;
```
Comment (lucferbux) [inline frontend/src/pages/projects/screens/detail/DetailsSection.tsx:15]: My question bellow is related to this, I'm not sure why we need to convert the id to a general string, the hole power of enums is that you limit the scope of the ids, in this case the ones defined in the enum, shouldn't we stick to this type here?

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: frontend/src/pages/projects/screens/detail/DetailsSection.tsx (hunk 3 above)
Comment (Gkrumbach07) [reply frontend/src/pages/projects/screens/detail/DetailsSection.tsx:15]: Same change was made in this PR
https://github.com/opendatahub-io/odh-dashboard/pull/983#discussion_r1126748763

I would do the same thing here @jenny-s51 that you did in the comment linked

File: frontend/src/pages/projects/screens/projects/EmptyProjects.tsx (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/pages/projects/screens/projects/EmptyProjects.tsx:30]: unless there is a `<Link ... />` component inside the button. In that case i dont think you need component='a' 

@jenny-s51 please confirm if this is the case

File: frontend/src/pages/projects/screens/projects/ManageProjectModal.tsx [hunk 4]
```diff
@@ -86,8 +86,6 @@ const ManageProjectModal: React.FC<ManageProjectModalProps> = ({
         </Button>,
       ]}
     >
-      <Stack hasGutter>
-        <StackItem>
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/screens/projects/ManageProjectModal.tsx:90]: I think @andrewballantyne mentioned this in another PR. The Stack adds spacing to elelnts that we are relying on, so i wouldn't remove it.

Is there an a11y issue with the stack?

**Review by Gkrumbach07 — COMMENTED** [review]
Added some comments. I think there are some resolutions we made in other PRs that apply in this PR

File: frontend/src/pages/projects/screens/detail/DetailsSection.tsx (hunk 3 above)
Comment (jenny-s51) [reply frontend/src/pages/projects/screens/detail/DetailsSection.tsx]: Rebased - this file should be up to date now

File: frontend/src/pages/projects/screens/projects/EmptyProjects.tsx (hunk 1 above)
Comment (jenny-s51) [reply frontend/src/pages/projects/screens/projects/EmptyProjects.tsx:30]: You are correct @Gkrumbach07 - reverted in https://github.com/opendatahub-io/odh-dashboard/pull/983/commits/fbad4010e0eefada39d8a983144c99babcfcd033

**Review by Gkrumbach07 — APPROVED** [review]
Table changes are inline with the other PR. Everything looks good to me. And I think the last of the comments are resolved (or can be resolved imo).

/lgtm

File: frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTokenInput.tsx [hunk 5]
```diff
@@ -45,6 +45,7 @@ const ServingRuntimeTokenInput: React.FC<ServingRuntimeTokenInputProps> = ({
   return (
     <FormGroup
       label="Service account name"
+      fieldId="simple-form-name-01"
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTokenInput.tsx:48]: Is there any reason we set this field ID to `simple-form-name-01`? That looks like a random general testing ID.

File: frontend/src/pages/projects/screens/spawner/environmentVariables/EnvTypeSelectField.tsx [hunk 6]
```diff
@@ -36,6 +36,7 @@ const EnvTypeSelectField: React.FC<EnvTypeSelectFieldProps> = ({
               onToggle={() => setOpen(!open)}
               selections={envVariable.type || ''}
               placeholderText="Select one"
+              aria-label="Select one"
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/projects/screens/spawner/environmentVariables/EnvTypeSelectField.tsx:39]: Maybe change it to "Select the environment variable type"?

Comment (lucferbux) [reply frontend/src/pages/projects/screens/spawner/environmentVariables/EnvTypeSelectField.tsx:39]: +1

File: frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTokenInput.tsx (hunk 5 above)
Comment (jenny-s51) [reply frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTokenInput.tsx]: Updated this value to `service-account-form-name`

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Seems we still have a couple issues
image

**Pull request merged**